src/ring_theory/local_properties.lean

```diff
@@ -221,36 +221,59 @@ end properties
 
 section ideal
 
+open_locale non_zero_divisors
+
+/-- Let `I J : ideal R`. If the localization of `I` at each maximal ideal `P` is included in
+the localization of `J` at `P`, then `I ≤ J`. -/
+lemma ideal.le_of_localization_maximal {I J : ideal R}
+  (h : ∀ (P : ideal R) (hP : P.is_maximal),
+    ideal.map (algebra_map R (by exactI localization.at_prime P)) I ≤
+      ideal.map (algebra_map R (by exactI localization.at_prime P)) J) :
+  I ≤ J :=
+begin
+  intros x hx,
+  suffices : J.colon (ideal.span {x}) = ⊤,
+  { have := submodule.mem_colon.mp
+      (show (1 : R) ∈ J.colon (ideal.span {x}), from this.symm ▸ submodule.mem_top)
+      x (ideal.mem_span_singleton_self x),
+    simpa using this },
+  refine not.imp_symm (ideal.exists_le_maximal (J.colon (ideal.span {x}))) _,
+  push_neg,
+  unfreezingI { intros P hP le },
+  obtain ⟨⟨⟨a, ha⟩, ⟨s, hs⟩⟩, eq⟩ := (is_localization.mem_map_algebra_map_iff P.prime_compl _).mp
+    (h P hP (ideal.mem_map_of_mem _ hx)),
+  rw [← _root_.map_mul, ← sub_eq_zero, ← map_sub] at eq,
+  obtain ⟨⟨m, hm⟩, eq⟩ := (is_localization.map_eq_zero_iff P.prime_compl _ _).mp eq,
+  refine hs ((hP.is_prime.mem_or_mem (le (ideal.mem_colon_singleton.mpr _))).resolve_right hm),
+  simp only [subtype.coe_mk, sub_mul, sub_eq_zero, mul_assoc] at eq,
+  simpa only [eq, mul_comm] using J.mul_mem_right m ha
+end
+
+/-- Let `I J : ideal R`. If the localization of `I` at each maximal ideal `P` is equal to
+the localization of `J` at `P`, then `I = J`. -/
+theorem ideal.eq_of_localization_maximal {I J : ideal R}
+  (h : ∀ (P : ideal R) (hP : P.is_maximal),
+    ideal.map (algebra_map R (by exactI localization.at_prime P)) I =
+      ideal.map (algebra_map R (by exactI localization.at_prime P)) J) :
+  I = J :=
+le_antisymm
+  (ideal.le_of_localization_maximal (λ P hP, (h P hP).le))
+  (ideal.le_of_localization_maximal (λ P hP, (h P hP).ge))
+
+/-- An ideal is trivial if its localization at every maximal ideal is trivial. -/
+lemma ideal_eq_zero_of_localization' (I : ideal R)
+   (h : ∀ (J : ideal R) (hJ : J.is_maximal),
+      ideal.map (algebra_map R (by exactI (localization.at_prime J))) I = ⊥) : I = ⊥ :=
+ideal.eq_of_localization_maximal (λ P hP, (by simpa using h P hP))
+
 -- This proof should work for all modules, but we do not know how to localize a module yet.
```

Comment:
Maybe the API is not enough, but we have [is_localized_module](https://leanprover-community.github.io/mathlib_docs/algebra/module/localized_module.html#is_localized_module) and friends.

Reply:
True, the issue right now AFAIK is that we don't have anything on submodules in the localization. I'll update the comment appropriately.

Reply:
```suggestion
-- TODO: This proof should work for all modules, once we have enough material on submodules of
-- localized modules.
```

Reply:
For what it's worth, I tried to make sure the generalized proof should still apply to submodules.